src/tests/Interop/COM/NETClients/MiscTypes/Program.cs

```diff
@@ -0,0 +1,90 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Xunit;
+namespace NetClient
+{
+    using System;
+    using System.Runtime.InteropServices;
+
+    using TestLibrary;
+    using Xunit;
+    using Server.Contract;
+    using Server.Contract.Servers;
+
+    public unsafe class Program
+    {
+        [Fact]
+        public static int TestEntryPoint()
+        {
+            // RegFree COM is not supported on Windows Nano
+            if (TestLibrary.Utilities.IsWindowsNanoServer)
+            {
+                return 100;
+            }
+
+            try
+            {
+                RunTests();
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine($"Test object interop failure: {e}");
+                return 101;
+            }
+
+            return 100;
+        }
+
+        private static void RunTests()
```

Comment:
Should the rare types of VARIANT be covered in this PR? We would want to add test for them when trying to refactor.
Should we also test VARIANT types that we do not support? That should be a more verbose list.

Reply:
They can be, but I don't think there are necessarily needed in this PR. Once this is in we have a place to start adding them though. Do you have some specific ones you are immediately concerned about?

Reply:
I will add the negative case for user defined records (i.e., non `System.Guid`).

Reply:
They can be added later. The one I'm concerned most should be `System.Drawing.Color`, especially when dealing with multiple ALCs.